static/js/tools/map/chart_loader.tsx

```diff
@@ -291,102 +176,96 @@ export function ChartLoader(): JSX.Element {
       </div>
     );
   }
-  const facetInfo = getFacetInfo(
-    statVar.value,
-    rawData.allEnclosedPlaceStat,
-    rawData.metadataMap
-  );
 
-  const onPlay = async (callback: () => void) => {
-    setOnPlayCallback(() => callback);
-    const metaHash = statVar.value.metahash || BEST_AVAILABLE_METAHASH;
-    if (metaHash in sampleDatesChartData) {
-      callback();
-      return;
-    }
-    fetchData(
-      placeInfo.value,
-      statVar.value,
-      isLoading,
-      setRawData,
-      display.value.showTimeSlider,
-      allSampleDates,
-      allSampleDates[metaHash],
-      sampleDatesChartData,
-      setSampleDatesChartData
+  if (mapType === MAP_TYPE.D3 && chartStore.geoJson.error) {
+    return (
+      <div className="p-5">
+        {`Sorry, maps are not available for ${placeInfo.value.enclosedPlaceType} in ${placeInfo.value.selectedPlace.name}. Try picking another place or type of place.`}
+      </div>
     );
-  };
+  }
 
-  const updateDate = (date: string) => {
-    // Check if any data is fetched at all for the date
-    const metaHash = statVar.value.metahash || BEST_AVAILABLE_METAHASH;
-    let placeStatData = false;
-    if (
-      metaHash in sampleDatesChartData &&
-      date in sampleDatesChartData[metaHash]
-    ) {
-      for (const place in sampleDatesChartData[metaHash][date]
-        .enclosedPlaceStat) {
-        if (
-          sampleDatesChartData[metaHash][date].enclosedPlaceStat[place].value
-        ) {
-          placeStatData = true;
-          break;
-        }
-      }
-    }
-    statVar.setDate(date);
+  if (!_.isEqual(chartStore.mapValuesDates.context.statVar, statVar.value)) {
```

Comment:
maybe we can have a ready hook for mapValuesDates? or for rendering the chart? Just because there seems to be a lot of logic here about when to render a chart and we might want to extract that out. But maybe in a future PR

Reply:
Added the ready hooks for computed values.